balena/models/device.py

```diff
@@ -1033,7 +1033,41 @@ def set_to_release(self, uuid, commit_id):
         """
 
         device_id = self.get(uuid)['id']
-        release_id = self.release._Release__get_by_option('commit', commit_id)[0]['id'] if commit_id else None
+        release_id = self.release._Release__get_by_option('commit', commit_id, status='success')[0]['id'] if commit_id else None
```

Comment:
Don't we need to check that the returned array actually contains a results, and if that's not the case throw a `ReleaseNotFound` exception?
I might be missing something, but it seems that `get_by__options` only checks `release['d']`.

Reply:
the `release['d']` check will raise `ReleaseNotFound` in both `None` or empty so I think it's enough